lib/code_corps/stripe_service/stripe_connect_subscription_service.ex

```diff
@@ -37,12 +37,14 @@ defmodule CodeCorps.StripeService.StripeConnectSubscriptionService do
          {:ok, %User{} = user} <- get_user_with_preloads(user_id),
          {:ok, %User{}} <- UserCanSubscribe.validate(user)
     do
-      {:ok, %StripeConnectSubscription{} = subscription} = do_find_or_create(project, user, attributes)
-
-      ProjectService.update_project_totals(project)
-      DonationGoalsService.update_project_goals(project)
-
-      {:ok, subscription}
+      case do_find_or_create(project, user, attributes) do
+        {:ok, %StripeConnectSubscription{} = subscription} ->
+          ProjectService.update_project_totals(project)
+          DonationGoalsService.update_project_goals(project)
+          {:ok, subscription}
+        {:error, error} ->
+          {:error, error}
+      end
```

Comment:
I think the simpler solution here is to not do a `case` statement but instead simply move the `do_find_or_create` logic as it was up into the `with` and only call the other services and return `{:ok, subscription}` in the `do`, letting the failure of `Stripe.APIErrorResponse` be dealt with by the `else`.